### ISSUE TITLE
SQLite statements with ALTER TABLE are not parsed

### ISSUE THREAD

**Issue by caffeinatedgaze** [sqlfluff/sqlfluff#5732]
### Search before asking

- [X] I searched the [issues](https://github.com/sqlfluff/sqlfluff/issues) and found no similar issues.

### What Happened

I used sqlfluff to format my sqlite code and it failed to parse `ALTER` statements.

### Expected Behaviour

Parsing `ALTER` statements.

### Observed Behaviour

Regardless of the fact that `sqlite` is derived from `ansi` dialect code, parsing of `alter` statements fails. 

### How to reproduce

Sample SQL code that causes trouble.

```sql
ALTER TABLE message RENAME TO nachrichten;
```

Command that fails.

```bash
$ sqlfluff --dialect sqlite trouble.sql
RuntimeError: Grammar refers to the 'Modify' keyword which was not found in the sqlite dialect. 

```

### Dialect

sqlite and ansi

### Version

```
$ sqlfluff --version
sqlfluff, version 3.0.3

$ python3 --version
Python 3.8.18
```

### Configuration

Default. I only pass the dialect as a keyword flag.

### Are you willing to work on and submit a PR to address the issue?

- [X] Yes I am willing to submit a PR!

### Code of Conduct

- [X] I agree to follow this project's [Code of Conduct](https://github.com/sqlfluff/sqlfluff/blob/main/CODE_OF_CONDUCT.md)

**Comment by caffeinatedgaze** [comment]
This parsing error cannot be ignored by means other than `.sqlfluffignore`, which is an overkill for me. Steps to reproduce:

```sh
echo 'ALTER TABLE messages ADD author_id INTEGER NULL REFERENCES users (id); -- noqa: PRS' | sqlfluff format --dialect sqlite -
```

**Comment by keraion** [comment]
Closed by #5888
